logstash-core/lib/logstash/outputs/base.rb

```diff
@@ -20,42 +20,40 @@ class LogStash::Outputs::Base < LogStash::Plugin
 
   # The codec used for output data. Output codecs are a convenient method for encoding your data before it leaves the output, without needing a separate filter in your Logstash pipeline.
   config :codec, :validate => :codec, :default => "plain"
+  
+  # Set or return concurrency type
+  def self.concurrency(type=nil)
+    if type
+      @concurrency = type
+
+      # TODO remove this in Logstash 6.0
+      # when we no longer support the :legacy_worker type
+      # This is hacky, but it can only be here
+      if type == :legacy_worker
```

Comment:
Feels inconsistent to have them called `shared`, `single`, and then a two-word name `legacy worker`. Can we call it `legacy` instead of `legacy_worker`? What's the story on the `_worker` suffix?